### ISSUE TITLE
Implicit exposure handling

### ISSUE THREAD

**Issue by nicooga** [elixir-maru/maru_entity#15]
I think it's positive to discuss whether we should implement the same magic behavior grape has for declaring exposures. In grape, when declaring an exposure, if the class has an instance method with the same name it automatically uses it to serialize that exposure:

~~~ruby
class PostEntity < Grape::Entity
  expose :title

  private

  def title
    some_magic_code
  end
end
~~~

This way I could go from this:

~~~elixir
expose :rating_count, [if: &should_show_metrics?/2], &rating_count/2
expose :rating_value, [if: &should_show_metrics?/2], &rating_value/2
expose :max_price,    [if: &should_show_metrics?/2], &max_price/2
expose :min_price,    [if: &should_show_metrics?/2], &min_price/2
expose :image,        [if: &should_show_metrics?/2], &image/2
~~~

... to this:

~~~elixir
expose ~w[rating_count rating_value max_price min_price image]a, if: &should_show_metrics?/2
~~~

... which clearly is a win win in terms of LOC and code readability. The downside is that we would be going for the magic, implicit side, when elixir in general is about being explicit.

**Comment by falood** [comment]
Hi @nicooga 
I think about this issue again and again, but I still don't think the ruby-like-grammar  is good for elixir.
In ruby, developer don't care `variable` and `method`, when we call `dog.weight` after `dog = Dog.new`, `weight` can be either a method or an attribute. 
But it seems it's not promoted in elixir, we should use explicit `()` after function even if it's `func/0`
And for myself, two different behaviors for the same grammar make me confuse, I don't know `expose :a` means `Map.get(item, :a)` or `__MODULE__.a(item, options)`

Sorry I can't find a DSL I think is better for your use case for now, let's keep discussing about it 🙂

**Comment by nicooga** [comment]
I agree, Maybe we can circumvent the code repeating problem using `with_options`:

~~~elixir
with_options if: &should_show_metrics?/2 do
  expose :rating_count, &rating_count/2
  expose :rating_value, &rating_value/2
  expose :max_price, &max_price/2
  expose :min_price, &min_price/2
  expose :image, &image/2
end
~~~

This also is a feature present on grape with which people will be familiar.
